### ISSUE TITLE
[Feature Request]Add more comm op support on gpu_hlo_cost_analysis

### ISSUE THREAD

**Issue by zjjott** [openxla/xla#11934]
such as allgather/reducescatter/alltoall, which is commonly usely on fsdp/moe model
my suggestion:
1. decoupling `output_bytes_accessed` and `bytes_accessed`,which `bytes_accessed` shoud be operand size bytes. not put them together
2. add allgather/reducescatter/alltoall support
3. support upper op  on `gpu_collective_performance_model.cc`;

we can make a table to show how many bytes will send inter nodes and inner nodes

I have prepared a PR for them, is this Feature Request needed?

additional question:
Can I get which gpu is inner node gpu?  use nvml api?

**Comment by zjjott** [comment]
@Tixxx @olegshyshkov

**Comment by olegshyshkov** [comment]
Could you please add more details why the current model doesn't work for comm ops?

> decoupling output_bytes_accessed and bytes_accessed,which bytes_accessed shoud be operand size bytes. not put them together

There is also [`operand_bytes_accessed`](https://github.com/openxla/xla/blob/8e4bb45c02c5e7c34ce039839c7241441ef03233/xla/service/hlo_cost_analysis.h#L283). And `bytes_accessed` is just `sum(operand_bytes_accessed) + output_bytes_accessed`. 

We don't use `bytes_accessed` in [gpu_performance_model.cc](https://github.com/openxla/xla/blob/b3304688637b6cdcd5b016d3044b4707b06b15d7/xla/service/gpu/model/gpu_performance_model.cc), but I see uses in other parts of the codebase and I can't predict all the implications if we change semantics of the field.

**Comment by zjjott** [comment]
@olegshyshkov OK,I understand~
so the second feature, do we need  more op support  like allgather/reducescatter/alltoall cost analysis? I have finish a PR to do this

**Comment by Tixxx** [comment]
> @olegshyshkov OK,I understand~ so the second feature, do we need more op support like allgather/reducescatter/alltoall cost analysis? I have finish a PR to do this

Yes, having more support for those ops will be good if you can add it.

**Comment by penumallas1261** [comment]
Hi @zjjott ,

Could you please confirm if this issue is resolved for you ? Please feel free to close the issue if it is resolved ?

**Comment by penumallas1261** [comment]
Hi @zjjott ,

Closing the issue due to inactivity for over 3 months, Please feel free to reopen the issue if you still have a concern. Thanks!
